src/components/App.js

```diff
@@ -1,14 +1,41 @@
+import React from 'react';
 import Display from './Display';
 import Buttonspanel from './ButtonsPanel';
+import calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <>
-      <Display result="0" />
-      <Buttonspanel />
-    </>
+export default class APP extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
```

Comment:
### Improve the `division by zero` functionality 
#### Consider this an edge case.
-  As a user, I am unable to check if I can divide by zero. Instead of having `REACT `scream at the user, wouldn't it be gracious to have a custom error message to warn the user the operation is not possible? What do you think?
![zero-error](https://user-images.githubusercontent.com/22345961/109384589-e8576500-78fe-11eb-999f-46cff96ea851.png)

## Improve the `negative` operation 
- [ ] As a user when I type the `-` and a number then click the `=` operation, I expect to see the result of the negated number on the display. Instead, this is what I am met with. Would you mind sorting this out?
![negaative-error](https://user-images.githubusercontent.com/22345961/109384623-310f1e00-78ff-11eb-9176-07e02a6af3cd.png)